backend/routes/forms/form.py

```diff
@@ -70,10 +75,17 @@ async def patch(self, request: Request) -> JSONResponse:
         except json.decoder.JSONDecodeError:
             return JSONResponse("Expected a JSON body.", 400)
 
-        form_id = {"_id": request.path_params["form_id"].lower()}
-        if raw_form := await request.state.db.forms.find_one(form_id):
+        form_id = request.path_params["form_id"].lower()
+
+        try:
+            if not await discord.verify_edit_access(form_id, request):
+                return JSONResponse({"error": "unauthorized"}, status_code=401)
+        except discord.FormNotFoundError:
```

Comment:
Doesn't Starlette allow you to raise exceptions to indicate responses? We could have `FormNotFoundError` be a subclass of that and then just let it propagate.

Reply:
0344d9a